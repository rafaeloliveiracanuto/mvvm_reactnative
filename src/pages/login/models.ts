import React from 'react'

export interface LoginViewModel {
  email: string
  password: string
  setPassword: React.Dispatch<React.SetStateAction<string>>
  setEmail: React.Dispatch<React.SetStateAction<string>>
  isLoading: boolean
  onSubmit: () => void
}