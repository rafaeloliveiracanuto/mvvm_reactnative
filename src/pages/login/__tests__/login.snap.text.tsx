import LoginView from '../view'
import { LoginViewModel } from '../models'
import React from 'react'

import renderer from 'react-test-renderer'

jest.mock('../view.model', () => {
  const fnLoginViewModel = jest.fn()
  fnLoginViewModel.mockImplementation(
    () => ({
      email: '',
      setEmail: jest.fn(),
      isLoading: false,
      onSubmit: jest.fn(),
      password: '',
      setPassword: jest.fn(),
    } as LoginViewModel)
  )

  return fnLoginViewModel
})

import useLoginViewModel from '../view.model'

it('should render correctly', () => {
  const { toJSON } = renderer.create(<LoginView />)
  expect(toJSON()).toMatchSnapshot()
})

it('should render correctly with email', () => {
  useLoginViewModel.mockImplementation(
    () => ({
      email: 'rafael@gmail.com',
      setEmail: jest.fn(),
      isLoading: false,
      onSubmit: jest.fn(),
      password: '',
      setPassword: jest.fn(),
    } as LoginViewModel)
  )

  const { toJSON } = renderer.create(<LoginView />)
  expect(toJSON()).toMatchSnapshot()
})

it('should render correctly with password', () => {
  useLoginViewModel.mockImplementation(
    () => ({
      email: '',
      setEmail: jest.fn(),
      isLoading: false,
      onSubmit: jest.fn(),
      password: '12345678',
      setPassword: jest.fn(),
    } as LoginViewModel)
  )
    
  const { toJSON } = renderer.create(<LoginView />)
  expect(toJSON()).toMatchSnapshot()
})

it('should render correctly with email and password', () => {
  useLoginViewModel.mockImplementation(
    () => ({
      email: 'rafael@gmail.com',
      setEmail: jest.fn(),
      isLoading: false,
      onSubmit: jest.fn(),
      password: '12345678',
      setPassword: jest.fn(),
    } as LoginViewModel)
  )
      
  const { toJSON } = renderer.create(<LoginView />)
  expect(toJSON()).toMatchSnapshot()
})

it('should render correctly while is loading', () => {
  useLoginViewModel.mockImplementation(
    () => ({
      email: 'rafael@gmail.com',
      setEmail: jest.fn(),
      isLoading: true,
      onSubmit: jest.fn(),
      password: '12345678',
      setPassword: jest.fn(),
    } as LoginViewModel)
  )
        
  const { toJSON } = renderer.create(<LoginView />)
  expect(toJSON()).toMatchSnapshot()
})

