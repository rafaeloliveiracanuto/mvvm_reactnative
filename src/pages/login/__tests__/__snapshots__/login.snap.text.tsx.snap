// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="rafael@email.com"
      value=""
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="******"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly while is loading 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="rafael@email.com"
      value="rafael@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="******"
      value="12345678"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with email 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="rafael@email.com"
      value="rafael@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="******"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with email and password 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="rafael@email.com"
      value="rafael@gmail.com"
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="******"
      value="12345678"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should render correctly with password 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="rafael@email.com"
      value=""
    />
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="******"
      value="12345678"
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
